packages/world/src/systems/MineSystem.sol

```diff
@@ -251,16 +252,21 @@ contract MineSystem is System {
     }
   }
 
-  function _handleDrop(EntityId caller, EntityId mined, ObjectType minedType, Vec3 coord) internal {
+  function _handleDrop(EntityId caller, EntityId mined, ObjectType minedType, Vec3 minedCoord) internal {
     // Get drops with all metadata for resource tracking
-    ObjectAmount[] memory result = RandomResourceLib._getMineDrops(mined, minedType, coord);
+    ObjectAmount[] memory result = RandomResourceLib._getMineDrops(mined, minedType, minedCoord);
 
     for (uint256 i = 0; i < result.length; i++) {
       (ObjectType dropType, uint128 amount) = (result[i].objectType, result[i].amount);
 
       if (amount == 0) continue;
 
-      InventoryUtils.addObject(caller, dropType, amount);
+      try InventoryUtils.addObject(caller, dropType, amount) {
```

Comment:
yo what you can try catch in solidity???

Reply:
can you use it when calling another contract/

Reply:
Yes, but there are some limitations. It doesn't really catch all issues, for example if there is an issue when decoding the return data, as that happens in the caller contract it will still revert